conftest.py

```diff
@@ -193,6 +193,12 @@ def update_env(node, modules=None, cflags=None, port=None, termflags=None):
         node.env['USEMODULE'] = ' '.join(str(m) for m in modules)
     elif modules is not None:
         node.env['USEMODULE'] = modules
+    if "USEMODULE" in node.env and \
+       os.environ.get('BUILD_IN_DOCKER', 0) == '1':
+        # workaround to inject USEMODULE into docker container
+        # see: https://github.com/RIOT-OS/RIOT/issues/14504
+        node.env['DOCKER_ENVIRONMENT_CMDLINE'] = "-e 'USEMODULE={}'" \
+            .format(node.env['USEMODULE'])
```

Comment:
Mimics current handling done when launching from command line, not ideal, not intuitive but its the current way to do it, maybe it can be eventually fixed in master and this patch removed, but for now its the way to go

Reply:
I just realized: `DOCKER_ENVIRONMENT_CMDLINE` might be already set with something by the user. Will change, so it is amended if set.

Reply:
Done.